EFCore.InMemory.HierarchyId/Storage/InMemoryHierarchyIdTypeMappingSourcePlugin.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Microsoft.EntityFrameworkCore.InMemory.Storage.Internal;
+using Microsoft.EntityFrameworkCore.Storage;
+
+namespace Microsoft.EntityFrameworkCore.InMemory.Storage
+{
+    internal class InMemoryHierarchyIdTypeMappingSourcePlugin : ITypeMappingSourcePlugin
+    {
+        public CoreTypeMapping FindMapping(in TypeMappingInfo mappingInfo)
+        {
+            var clrType = mappingInfo.ClrType;
+
+            var valueComparer = new ValueComparer<HierarchyId>(
+                (w1, w2) => w1.Equals(w2),
+                w => w.GetHashCode()
+            );
+
+            return typeof(HierarchyId).IsAssignableFrom(clrType)
+                ? new InMemoryTypeMapping(clrType, valueComparer, null)
```

Comment:
We should also make this non-internal in efcore?

Reply:
Yes, yes, yes, This isn't the first time I hit this. In the past, I just re-created a similar class it to avoid breaks. We should probably do the same here.